src/main.c

```diff
@@ -49,7 +54,7 @@ static void set_own_dir(const char *argv0) {
 	size_t l = strlen(argv0);
 	while(l && argv0[l - 1] != '/')
 		l--;
-	if(l == 0)
+	if(l == 0 || l >= sizeof(own_dir))
```

Comment:
could you put this into a sep. commit pls ? this seems unrelated to the change at hand